habitat_baselines/utils/common.py

```diff
@@ -594,3 +630,72 @@ def action_to_velocity_control(
         }
     }
     return step_action
+
+
+def is_continuous_action_space(action_space) -> bool:
+    if not isinstance(action_space, spaces.Dict):
+        return False
+
+    for v in action_space.spaces.values():
+        if isinstance(v, spaces.Dict):
+            return is_continuous_action_space(v)
+        elif isinstance(v, spaces.Box):
+            return True
+
+    return False
+
+
+def get_num_actions(action_space) -> int:
+    queue = [action_space]
+    num_actions = 0
+    while queue:
+        v = queue.pop()
+        if isinstance(v, spaces.Dict):
+            for v in v.spaces.values():
+                queue.append(v)
+        elif isinstance(v, spaces.Box):
+            num_actions += v.shape[0]
+        else:
+            num_actions += 1
+
+    return num_actions
+
+
+def action_array_to_dict(
```

Comment:
I don't understand what this is. I don't see it used anywhere. I hope it's not on a perf-critical path like running after every DNN inference for all envs. In my mind, actions are just a big flat list of floats coming out of a DNN and piped directly into the Gala sim, so I'm surprised at all this complexity.